controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -455,8 +455,16 @@ private ApplicationRevision toApplicationPackageRevision(ApplicationPackage appl
     private void registerRotationInDns(Rotation rotation, String dnsName) {
         try {
             Optional<Record> record = nameService.findRecord(Record.Type.CNAME, RecordName.from(dnsName));
-            if (!record.isPresent()) {
-                RecordId id = nameService.createCname(RecordName.from(dnsName), RecordData.from(rotation.name()));
+            RecordData rotationName = RecordData.fqdn(rotation.name());
+            if (record.isPresent()) {
+                // Ensure that the existing record points to the correct rotation
+                if (!record.get().data().equals(rotationName)) {
+                    nameService.updateRecord(record.get().id(), rotationName);
```

Comment:
Should we start with just logging here before enabling it?

Reply:
I think it will be fine. I have test case in CD I'll verify on next run.